gtsam_unstable/slam/SmartRangeFactor.h

```diff
@@ -105,8 +106,8 @@ class SmartRangeFactor: public NoiseModelFactor {
     }
 
     Circle2 circle1 = circles.front();
-    boost::optional<Point2> best_fh;
-    auto bestCircle2 = boost::make_optional(false, circle1);  // fixes issue #38
+    std::optional<Point2> best_fh;
+    std::optional<Circle2> bestCircle2 = std::nullopt;  // fixes issue #38
```

Comment:
Sure about this?

Reply:
```
template<class T> inline optional<T> make_optional ( bool condition, T const& v ) ;  [new in 1.34]

```
This signature will set the optional variable to `v` if `condition` is true. Here it is hardcoded to false. That's why I did this. Also there is no equivalent in std::make_optional.